albatross/core/traits.h

```diff
@@ -28,36 +28,82 @@ namespace albatross {
  */
 template <class T> struct delay_static_assert : std::false_type {};
 
+/*
+ * In CovarianceFunction we frequently inspect for definitions of
+ * call_impl_ which MUST be defined for const references to objects
+ * (so that repeated covariance matrix evaluations return the same thing
+ *  and so the computations are not repeatedly copying.)
+ * This type conversion utility will turn a type `T` into `const T&`
+ */
+template <class T> struct call_impl_arg_type {
+  typedef
+      typename std::add_lvalue_reference<typename std::add_const<T>::type>::type
+          type;
+};
+
 /*
  * This determines whether or not a class has a method defined for,
  *   `operator() (X x, Y y, Z z, ...)`
```

Comment:
You updated other comments with the new const reference types, shouldn't these be updated as well?

Reply:
good catch